features/admin.server-configurations.v1/forms/password-recovery-form/password-recovery-form.tsx

```diff
@@ -0,0 +1,709 @@
+ * under the License.
+ */
+
+import Chip from "@oxygen-ui/react/Chip";
+import {
+    PasswordRecoveryConfigurationFormPropsInterface,
+    PasswordRecoveryFormConstants,
+    PasswordRecoveryFormErrorValidationsInterface,
+    PasswordRecoveryFormUpdatableConfigsInterface,
+    PasswordRecoveryFormValuesInterface } from "@wso2is/admin.server-configurations.v1";
+import { CommonUtils } from "@wso2is/core/utils";
+import { Field, Form } from "@wso2is/form";
+import { Heading, Hint } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, ReactNode, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { Divider, Label } from "semantic-ui-react";
+import { AppState } from "../../../admin.core.v1";
+import { GovernanceConnectorConstants } from "../../constants/governance-connector-constants";
+import { ServerConfigurationsConstants } from "../../constants/server-configurations-constants";
+import { ConnectorPropertyInterface } from "../../models/governance-connectors";
+import "./password-recovery-form.scss";
+
+const FORM_ID: string = "governance-connectors-password-recovery-form";
+
+/**
+ * Password Recovery Configuration Form.
+ *
+ * @param props - Props injected to the component.
+ * @returns Functional component.
+ */
+export const PasswordRecoveryConfigurationForm: FunctionComponent<PasswordRecoveryConfigurationFormPropsInterface> = (
+    props: PasswordRecoveryConfigurationFormPropsInterface
+): ReactElement => {
+
+    const {
+        initialValues,
+        onSubmit,
+        readOnly,
+        isConnectorEnabled,
+        isSubmitting,
+        ["data-componentid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+    const [ initialConnectorValues, setInitialConnectorValues ]
+        = useState<PasswordRecoveryFormValuesInterface>(undefined);
+    const [ isEmailRecoveryEnabled, setIsEmailRecoveryEnabled ] = useState<boolean>(false);
+    const [ isSMSRecoveryEnabled, setIsSMSRecoveryEnabled ] = useState<boolean>(false);
+    const [ isUpperCaseEnabled, setIsUpperCaseEnabled ] = useState<boolean>(false);
+    const [ isLowerCaseEnabled, setIsLowerCaseEnabled ] = useState<boolean>(false);
+    const [ isNumericEnabled, setIsNumericEnabled ] = useState<boolean>(false);
+    const showSmsOtpPwdRecoveryFeatureStatusChip: boolean =
+        useSelector((state: AppState) => state?.config?.ui?.showSmsOtpPwdRecoveryFeatureStatusChip);
+
+    /**
+     * Flattens and resolved form initial values and field metadata.
+     */
+    useEffect(() => {
+        if (isEmpty(initialValues?.properties)) {
+            return;
+        }
+
+        let resolvedInitialValues: PasswordRecoveryFormValuesInterface = null;
+
+        initialValues.properties.map((property: ConnectorPropertyInterface) => {
```

Comment:
```suggestion
        initialValues?.properties?.map((property: ConnectorPropertyInterface) => {
```

Reply:
`if (isEmpty(initialValues?.properties)) {
            return;
        }`

This code segment is there above the suggested code lines. Is this change needed?